src/search/ApproxSearchQuery.jl

```diff
@@ -0,0 +1,183 @@
+# Approxiamte Search
+# ==================
+#
+# Approximate sequence search tools.
+#
+# This file is a part of BioJulia.
+# License is MIT: https://github.com/BioJulia/BioSequences.jl/blob/master/LICENSE.md
+
+"""
+Query type for approximate sequence search.
+"""
+struct ApproximateSearchQuery{F<:Function,S<:BioSequence}
```

Comment:
Same as exact query: Would be nice with a simple docstring for this type, e.g. how to construct it, maybe even doctests.